engine-types/src/types/address.rs

```diff
@@ -0,0 +1,172 @@
+use crate::{format, String, TryFrom, H160};
+use borsh::maybestd::io;
+use borsh::{BorshDeserialize, BorshSerialize};
+
+/// Base Eth Address type
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub struct Address(H160);
+
+impl Address {
+    /// Construct Address from H160
+    pub const fn new(val: H160) -> Self {
+        Self(val)
+    }
+
+    /// Get raw H160 data
+    pub fn raw(&self) -> H160 {
+        self.0
+    }
+
+    /// Encode address to string
+    pub fn encode(&self) -> String {
+        hex::encode(self.0.as_bytes())
+    }
+
+    pub fn decode(address: String) -> Result<Address, error::AddressError> {
```

Comment:
```suggestion
    pub fn decode(address: &str) -> Result<Address, error::AddressError> {
```

I don't think ownership of the address string should be needed here, so can pass by reference instead.